R/VarSelection.R

```diff
@@ -194,15 +196,25 @@ select_cov <- function(nsdm_finput, scale, ClimaticVariablesBands,
 
   species_name <- paste0("SpeciesData.XY.", scale)
   background_name <- paste0("Background.XY.", scale)
+  absences_name <- paste0("Absences.XY.", scale)
+
 
   # Select the best subset of covariates for each species using covsel package
-  myResp.xy <- rbind(nsdm_finput[[species_name]], nsdm_finput[[background_name]])
-  row.names(myResp.xy) <- c(1:nrow(myResp.xy))
-  myResp <- as.vector(c(rep(1, nrow(nsdm_finput[[species_name]])), rep(0, nrow(nsdm_finput[[background_name]]))))
-  myExpl.covsel <- terra::extract(IndVar, myResp.xy, rm.na=TRUE, df=TRUE)[, -1]
+  if(!is.null(nsdm_finput[[background_name]])) {
+    myResp.xy <- rbind(nsdm_finput[[species_name]], nsdm_finput[[background_name]])
+  } else {
+    myResp.xy <- rbind(nsdm_finput[[species_name]], nsdm_finput[[absences_name]])
+  }
+  row.names(myResp.xy)<-c(1:nrow(myResp.xy))
+  if(!is.null(nsdm_finput[[background_name]])) {
+    myResp <- as.vector(c(rep(1,nrow(nsdm_finput[[species_name]])),rep(0,nrow(nsdm_finput[[background_name]]))))
```

Comment:
This can be simply written as `c(rep(1,nrow(nsdm_finput[[species_name]])),rep(0,nrow(nsdm_finput[[background_name]])))`